src/roles/httpd/templates/foreman-ssl-vhost.conf.j2

```diff
@@ -63,10 +63,64 @@
   ProxyPass /pulp/assets/ {{ httpd_pulp_api_backend }}/pulp/assets/
   ProxyPassReverse /pulp/assets/ {{ httpd_pulp_api_backend }}/pulp/assets/
 
+  ## Foreman public assets - served directly from filesystem
+  DocumentRoot "{{ httpd_foreman_public_dir }}"
+
+  <Directory "{{ httpd_foreman_public_dir }}">
+    Options SymLinksIfOwnerMatch
+    AllowOverride None
+    Require all granted
+  </Directory>
+
+  ## Serve precompiled gzipped assets
+  <FilesMatch \.css\.gz$>
+    ForceType text/css
+    Header set Content-Encoding gzip
+    SetEnv no-gzip
+  </FilesMatch>
+
+  <FilesMatch \.js\.gz$>
+    ForceType text/javascript
+    Header set Content-Encoding gzip
+    SetEnv no-gzip
+  </FilesMatch>
+
+  <FilesMatch \.svg\.gz$>
+    ForceType image/svg+xml
+    Header set Content-Encoding gzip
+    SetEnv no-gzip
+  </FilesMatch>
+
+  ## Assets and webpack - caching and compression
+  <LocationMatch "^/(assets|webpack)">
+    Options SymLinksIfOwnerMatch
+    AllowOverride None
+    Require all granted
+
+    # Use standard http expire header for assets instead of ETag
+    <IfModule mod_expires.c>
+      Header unset ETag
+      FileETag None
+      ExpiresActive On
+      ExpiresDefault "access plus 1 year"
+    </IfModule>
+
+    # Return compressed assets if they are precompiled
+    RewriteEngine On
+    # Make sure the browser supports gzip encoding and file with .gz added
+    # does exist on disc before we rewrite with the extension
+    RewriteCond %{HTTP:Accept-Encoding} \b(x-)?gzip\b
+    RewriteCond %{REQUEST_FILENAME} \.(css|js|svg)$
+    RewriteCond %{REQUEST_FILENAME}.gz -s
+    RewriteRule ^(.+) $1.gz [L]
+  </LocationMatch>
+
   ## Proxy rules
   ProxyRequests Off
   ProxyPreserveHost On
   ProxyAddHeaders On
+  ProxyPass /assets !
+  ProxyPass /webpack !
```

Comment:
`/webpack` is basically a static assets folder. Why do we want to `ProxyPass` it? We can serve it directly from Apache too.
And the same thing goes for the assets folder.

Reply:
[`!` means "do not proxy"](https://httpd.apache.org/docs/current/mod/mod_proxy.html#proxypass): 
> The ! directive is useful in situations where you don't want to reverse-proxy a subdirectory, e.g.